frame/support/src/storage/generator/double_map.rs

```diff
@@ -356,12 +358,29 @@ impl<
 		}
 	}
 
+	fn iter_key_prefix(k1: impl EncodeLike<K1>) -> Self::KeyPrefixIterator {
+		let prefix = G::storage_double_map_final_key1(k1);
+		Self::KeyPrefixIterator {
+			prefix: prefix.clone(),
+			previous_key: prefix,
+			drain: false,
+			closure: |raw_key_without_prefix| {
+				let mut key_material = G::Hasher2::reverse(raw_key_without_prefix);
+				K2::decode(&mut key_material)
+			}
+		}
+	}
+
 	fn drain_prefix(k1: impl EncodeLike<K1>) -> Self::PrefixIterator {
 		let mut iterator = Self::iter_prefix(k1);
 		iterator.drain = true;
 		iterator
 	}
 
+	fn drain_key_prefix(k1: impl EncodeLike<K1>) -> Self::KeyPrefixIterator {
```

Comment:
Same thing here: if the only difference is the return type, but it still drains all keys and values, then I'd prefer to make people do `drain_prefix(k).map(|(k, _v)| k)`. It means that `KeyPrefixIterator` can be simpler.